src/main/java/org/jabref/gui/filelist/AttachFileAction.java

```diff
@@ -2,43 +2,61 @@
 
 import java.util.Optional;
 
+import org.jabref.Globals;
 import org.jabref.gui.BasePanel;
-import org.jabref.gui.actions.BaseAction;
+import org.jabref.gui.DialogService;
+import org.jabref.gui.actions.SimpleCommand;
 import org.jabref.gui.undo.UndoableFieldChange;
+import org.jabref.gui.util.FileDialogConfiguration;
 import org.jabref.logic.l10n.Localization;
 import org.jabref.model.FieldChange;
 import org.jabref.model.entry.BibEntry;
 import org.jabref.model.entry.FieldName;
 import org.jabref.model.entry.LinkedFile;
+import org.jabref.preferences.JabRefPreferences;
 
-public class AttachFileAction implements BaseAction {
+public class AttachFileAction extends SimpleCommand {
 
     private final BasePanel panel;
+    private final DialogService dialogService;
 
-    public AttachFileAction(BasePanel panel) {
+    public AttachFileAction(BasePanel panel, DialogService dialogService) {
         this.panel = panel;
+        this.dialogService = dialogService;
     }
 
     @Override
-    public void action() {
+    public void execute() {
         if (panel.getSelectedEntries().size() != 1) {
             panel.output(Localization.lang("This operation requires exactly one item to be selected."));
             return;
         }
         BibEntry entry = panel.getSelectedEntries().get(0);
-        LinkedFile flEntry = new LinkedFile("", "", "");
-        FileListEntryEditor editor = new FileListEntryEditor(flEntry, false, true,
-                panel.getBibDatabaseContext());
-        editor.setVisible(true, true);
-        if (editor.okPressed()) {
-            Optional<FieldChange> fieldChange = entry.addFile(flEntry);
-
-            if (fieldChange.isPresent()) {
-                UndoableFieldChange ce = new UndoableFieldChange(entry, FieldName.FILE,
-                        entry.getField(FieldName.FILE).orElse(null), fieldChange.get().getNewValue());
-                panel.getUndoManager().addEdit(ce);
-                panel.markBaseChanged();
-            }
-        }
+
+        FileDialogConfiguration fileDialogConfiguration = new FileDialogConfiguration.Builder()
+                                                                                               .withInitialDirectory(Globals.prefs.get(JabRefPreferences.WORKING_DIRECTORY))
+                                                                                               .build();
+
+        dialogService.showFileOpenDialog(fileDialogConfiguration).ifPresent(newFile -> {
+            LinkedFile newLinkedFile = new LinkedFile("", newFile.toString(), "");
+
+            LinkedFilesWrapper wrapper = new LinkedFilesWrapper();
+            wrapper.setLinkedFile(newLinkedFile);
+            LinkedFileEditDialogView dialog = new LinkedFileEditDialogView(wrapper);
```

Comment:
Do we really want to show the linked file editor dialog here? In my experience, I just click "OK" anyway and would expect that this is the case for most users (recall how long it took until somebody complained that the description was not shown in the file list). At least, I would add a preferences that controls the appearance of the dialog.

Reply:
Previously there wasn't event he FileChooser shown .Instead it only showed the LinkedFilesEditor in which I then had to click open

Reply:
Ok, then was the previous behavior even worse. What advantage do you see of displaying the dialog every time?

Reply:
You can add a description and or choose a different file.

Reply:
If you really think that the majority of users wants to add a description, then keep the current behavior....